doc/api/n-api.md

```diff
@@ -3155,6 +3156,81 @@ support it:
 * If the function is not available, provide an alternate implementation
   that does not use the function.
 
+## Promises
+
+Promises facilitate the construction of chains of asynchronous actions such
```

Comment:
This is broad and I'm not sure it's very helpful to users.

I prefer a more informational tone about what promises _are_ in N-API and not what they enable users to do.

"A promise represents an eventual value - the result of an asynchronous action. A promise can be `await`ed on JavaScript's side or a callback could be attached to it".

The chaining aspect is interesting, but not really related to n-api from what I know.